dev/com.ibm.ws.microprofile.config.1.4_fat/fat/src/com/ibm/ws/microprofile/config14/test/BadObserverTest.java

```diff
@@ -0,0 +1,94 @@
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.ws.microprofile.config.fat.repeat.RepeatConfigActions;
+import com.ibm.ws.microprofile.config14.test.apps.badobserver.BadObserverServlet;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.rules.repeater.RepeatTests;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+/**
+ * <li> Application packaging is done in the @BeforeClass, instead of ant scripting.
+ * <li> Injects servers via @Server annotation. Annotation value corresponds to the
+ * server directory name in 'publish/servers/%annotation_value%' where ports get
+ * assigned to the LibertyServer instance when the 'testports.properties' does not
+ * get used.
+ * <li> Specifies an @RunWith(FATRunner.class) annotation. Traditionally this has been
+ * added to bytecode automatically by ant.
+ * <li> Uses the @TestServlet annotation to define test servlets. Notice that no @Test
+ * methods are defined in this class. All of the @Test methods are defined on the test
+ * servlet referenced by the annotation, and will be run whenever this test class runs.
+ */
+@RunWith(FATRunner.class)
+@Mode(TestMode.FULL)
+public class BadObserverTest extends FATServletClient {
+
+    public static final String APP_NAME = "badObserverApp";
+    public static final String SERVER_NAME = "Config14Server";
+
+    @ClassRule
+    public static RepeatTests r = RepeatConfigActions.repeatConfig14(SERVER_NAME);
+
+    @Server(SERVER_NAME)
+    @TestServlet(servlet = BadObserverServlet.class, contextRoot = APP_NAME)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        // Create a WebArchive that will have the file name 'app1.war' once it's written to a file
+        // Include the 'app1.web' package and all of it's java classes and sub-packages
+        // Automatically includes resources under 'test-applications/APP_NAME/resources/' folder
+        // Exports the resulting application to the ${server.config.dir}/apps/ directory
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addPackages(true, BadObserverServlet.class.getPackage());
+
+        ShrinkHelper.exportDropinAppToServer(server, war, DeployOptions.SERVER_ONLY);
+
+        server.startServer(true, false);//Don't validate, the app is going to throw a DeploymentException
+    }
+
+    @Test
+    public void testBadObserver() throws Exception {
+        List<String> msgs = server.findStringsInLogs("java.util.NoSuchElementException: CWMCG0015E: The property DOESNOTEXIST was not found in the configuration.");
```

Comment:
`java.util.NoSuchElementException` seems an odd thing to get out in the logs...

Reply:
Why? it is the real error that is designed to lead to the DeploymentException

Reply:
I didn't realise it's the exception specified on the `Config` interface. It looked like something that was being produced as a side effect of some internal behaviour.